warehouse/oidc/views.py

```diff
@@ -194,12 +194,12 @@ def _invalid(errors):
         [
             caveats.OIDCPublisher(
                 oidc_publisher_id=str(publisher.id),
-                oidc_claims={"ref": claims.get("ref"), "sha": claims.get("sha")},
             ),
             caveats.ProjectID(project_ids=[str(p.id) for p in publisher.projects]),
             caveats.Expiration(expires_at=expires_at, not_before=not_before),
         ],
         oidc_publisher_id=publisher.id,
+        additional={"ref": claims.get("ref"), "sha": claims.get("sha")},
```

Comment:
Since we'll want to use `additional` for other things as well, let's group this under a unique-looking key:

```suggestion
        additional={"oidc": {"ref": claims.get("ref"), "sha": claims.get("sha")}},
```

(That means an additional step during retrieval as well.)